authapi/views.py

```diff
@@ -41,37 +41,120 @@ class OrganizationUsersViewSet(viewsets.ViewSet):
     organizations.'''
     def create(self, request, organization_pk=None):
         '''Add a user to an organization.'''
-        serializer = OrganizationUserSerializer(data=request.data)
+        serializer = ExistingUserSerializer(data=request.data)
         serializer.is_valid(raise_exception=True)
-        user = get_object_or_404(User, pk=serializer.data['user_id'])
+        user = serializer.validated_data.get('user_id')
         org = get_object_or_404(SeedOrganization, pk=organization_pk)
         org.users.add(user)
-        org.save()
         return Response(status=status.HTTP_204_NO_CONTENT)
 
     def destroy(self, request, pk=None, organization_pk=None):
         '''Remove a user from an organization.'''
         user = get_object_or_404(User, pk=pk)
         org = get_object_or_404(SeedOrganization, pk=organization_pk)
         org.users.remove(user)
-        org.save()
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
 class TeamViewSet(viewsets.ModelViewSet):
     queryset = SeedTeam.objects.all()
     serializer_class = TeamSerializer
 
+    def get_queryset(self):
+        '''We want to still be able to modify archived organizations, but they
+        shouldn't show up on list views.
+
+        We have an archived query param, where 'true' shows archived, 'false'
+        omits them, and 'both' shows both.
+
+        We also have the query params permission_contains and object_id, which
+        allow users to filter the teams based on the permissions they
+        contain.'''
+        queryset = self.queryset
+        if self.action == 'list':
+            archived = self.request.query_params.get(
+                'archived', 'false').lower()
+            if archived == 'true':
+                queryset = queryset.filter(archived=True)
+            if archived == 'false':
+                queryset = queryset.filter(archived=False)
```

Comment:
Yes I thought about that, maybe we should be returning an error if it's not one of `true`, `false`, or `both`?